pkg/controllers/management/clusterstats/statsaggregator.go

```diff
@@ -220,3 +221,20 @@ func (s *StatsAggregator) machineChanged(key string, machine *v3.Node) (runtime.
 	}
 	return nil, nil
 }
+
+func determineNodeRole(machine *v3.Node) {
```

Comment:
I'm thinking maybe this logic should go where the role is usually set, nodessyncer. I wouldn't expect statsaggregator to set the role. Another possible option would be to not set it but infer that it is a worker here and have the determineNodeRole function return a bool.

Reply:
Agree. Also in that same line of thinking I'm not sure we even want to update the spec on a node. Using status or a util method to check both seems cleanest.

Reply:
so it seems like its just that race condition between the states getting updated prior to nodesync happening, so instead of updating the spec or lables / annotations, just going to be erroring if one if the specs is not set since that indicates the nodesync hasn't happened